GVFS/GVFS.Common/NuGetUpgrade/NuGetUpgrader.cs

```diff
@@ -117,6 +117,13 @@ public class NuGetUpgrader : ProductUpgrader
             if (!upgraderConfig.IsReady(out error))
             {
                 return false;
+            }
+
+            string gitBinPath = GVFSPlatform.Instance.GitInstallation.GetInstalledGitBinPath();
+            if (string.IsNullOrEmpty(GVFSPlatform.Instance.GitInstallation.GetInstalledGitBinPath()))

```

Comment:
Do we need to add a similar check in verbs and/or GVFS.Mount.exe?

I'm not sure if looking for git is part of our usual start up process or if we've been depending on the tracer throwing an exception to prevent the mount from succeeding.

Reply:
As we haven't had reports of this except for during upgrade, I was addressing the immediate specific issue.

From looking at the code, we don't have error handling around creating the tracer, so I don't think we depended on this behavior.. otherwise, I would have expected us to handle this error and exited gracefully.

If we need larger follow-up changes, I was suggesting we do that separately.

Reply:
> I was addressing the immediate specific issue

I should clarify my comment, I think the change in `NuGetUpgrader` is good.  The concern I had was with the more general change in `CreateTelemetryListeners`.

Today if a user tries to run VFS4G processes without git installed we get some automatic protection by virtue of the tracer requiring that `gitBinRoot` be set.  With the change to `CreateTelemetryListeners`, how (if at all) does the user experience change if they run VFS4G verbs (or GVFS.Mount.exe) without git installed?

If those processes don't fail right away there's a risk that GVFS.Mount.exe might stay mounted until some random time in the future when it ends up needing git to be present for whatever work it's doing (e.g. background tasks).

Reply:
We could also CreateTelemetryListeners to take a flag, 'failIfMissingGit'.  If we only want to change behavior for the upgrader.

Reply:
All Verbs that derive from [GVFSVerb.ForExistingEnlistment](https://github.com/Microsoft/VFSForGit/blob/master/GVFS/GVFS/CommandLine/GVFSVerb.cs#L1083), check that there is a git installation. This covers many, but not all verbs.

The other verbs are:


 Verb | Notes 
------ | ----------
`RepairVerb` | [Calls ](https://github.com/Microsoft/VFSForGit/blob/master/GVFS/GVFS/CommandLine/RepairVerb.cs#L42)`CreateWithoutRepoUrlFromDirectory`, which will [error ](https://github.com/jamill/VFSForGit/blob/d3b3c661ecb71851c53e7f0cc9e1ab2228d4c241/GVFS/GVFS.Common/Enlistment.cs#L18-L21)if it can't find git installation.
`CloneVerb` | Has a [built in check ](https://github.com/Microsoft/VFSForGit/blob/c9785b365a783db6fdf65429be1034e6ac7ee1a6/GVFS/GVFS/CommandLine/CloneVerb.cs#L296)
`LogVerb` and `ConfigVerb` and `UnmountVerb` |  do not construct a tracer.

Reply:
@jamill thanks for doing the leg work to confirm that VFS4G will check elsewhere that git is installed.